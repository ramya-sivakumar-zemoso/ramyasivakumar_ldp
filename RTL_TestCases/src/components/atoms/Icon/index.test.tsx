import { render, screen } from '@testing-library/react';
import MUIIcon from '.';
import '@testing-library/jest-dom';

describe('MUIIcon component', () => {
  test('renders with given children (icon name)', () => {
    render(<MUIIcon>home</MUIIcon>);
    const iconElement = screen.getByText('home');
    expect(iconElement).toBeInTheDocument();
    expect(iconElement.tagName.toLowerCase()).toBe('span');
  });

  test('applies MUI props correctly (color, fontSize)', () => {
    const { container } = render(
      <MUIIcon color="primary" fontSize="large">
        home
      </MUIIcon>
    );
    const iconElement = screen.getByText('home');
    expect(iconElement).toHaveClass('MuiIcon-colorPrimary');
    expect(iconElement).toHaveClass('MuiIcon-fontSizeLarge');
    expect(container.firstChild).toMatchSnapshot(); // Optional snapshot
  });

  test('passes arbitrary props (like aria-label)', () => {
    render(<MUIIcon aria-label="home-icon">home</MUIIcon>);
    const iconElement = screen.getByLabelText('home-icon');
    expect(iconElement).toBeInTheDocument();
  });

  test('renders correctly with different icons', () => {
    render(<MUIIcon>star</MUIIcon>);
    expect(screen.getByText('star')).toBeInTheDocument();
  });
});
