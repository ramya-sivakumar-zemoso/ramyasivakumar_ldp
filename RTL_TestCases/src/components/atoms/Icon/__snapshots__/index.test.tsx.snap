// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MUIIcon component > applies MUI props correctly (color, fontSize) 1`] = `
<span
  aria-hidden="true"
  class="material-icons notranslate MuiIcon-root MuiIcon-colorPrimary MuiIcon-fontSizeLarge css-v2t6vg-MuiIcon-root"
>
  home
</span>
`;
